### PR TITLE
refactor: rename SelfClientValidator to ConsensusHost and mv tests

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

Renames `SelfClientValidator` interface to `ConsensusHost` and moves associates tests to new test file.

part of: [#5315](https://github.com/cosmos/ibc-go/issues/5315)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/02-client/keeper/consensus_host.go
Comment (damiannolan) [inline modules/core/02-client/keeper/consensus_host.go:1]: I kinda feel like this should be in 07-tendermint. There should be no cycles as 02-client `keeper` pkg already imports `ibctm`.

The new keeper func default instantation would just be changed to:

```diff
return Keeper{
	storeKey:       key,
	cdc:            cdc,
	router:         router,
-	consensusHost:  NewTendermintConsensusHost(sk),
+	consensusHost:  ibctm.NewConsensusHost(sk),
	legacySubspace: legacySubspace,
	stakingKeeper:  sk,
	upgradeKeeper:  uk,
}
```

We could later remove the default instantiation and force a `types.ConsensusHost` impl as an arg to `NewKeeper` in a major release (panic if nil, similarly to other expected interfaces on node startup).

Comment (crodriguezvega) [reply modules/core/02-client/keeper/consensus_host.go:1]: Makes sense to move it to 07-tendermint. 👍

**Review by crodriguezvega — APPROVED** [review]
Looks good, Damian. I opened a [follow-up to do some renaming](https://github.com/cosmos/ibc-go/pull/6021), let me know what you think.

File: modules/core/02-client/keeper/consensus_host.go
Comment (damiannolan) [reply modules/core/02-client/keeper/consensus_host.go:1]: Great, I'll move it there then

**Review by charleenfei — APPROVED** [review]
thanks @damiannolan for doing this prep work to merge the feature branch in!

File: modules/light-clients/07-tendermint/consensus_host_test.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+				height = clienttypes.NewHeight(1, uint64(suite.chainA.GetContext().BlockHeight())-1)
+			},
+			expError: nil,
+		},
+		{
+			name: "latest height",
+			malleate: func() {
+				// historical info is set on BeginBlock in x/staking, which is now encapsulated within the FinalizeBlock abci method,
+				// thus, we do not have historical info for current height due to how the ibctesting library operates.
+				// ibctesting calls app.Commit() as a final step on NextBlock and we invoke test code before FinalizeBlock is called at the current height once again.
```
Comment (damiannolan) [inline modules/light-clients/07-tendermint/consensus_host_test.go:74]: Found some unexpected behaviour here with how testing lib operates. 

Invoking `TrackHistoricalInfo` directly resolves this for the time being.

Comment (damiannolan) [reply modules/light-clients/07-tendermint/consensus_host_test.go:74]: Opened https://github.com/cosmos/ibc-go/issues/6024

**Pull request merged**